pkg/network/tracer/tracer_classification_test.go

```diff
@@ -1750,6 +1752,61 @@ func testEdgeCasesProtocolClassification(t *testing.T, tr *Tracer, clientHost, t
 			teardown:   teardown,
 			validation: validateProtocolConnection(&protocols.Stack{Application: protocols.HTTP}),
 		},
+		{
+			// This test checks than we are not classification a connection as
+			// carrying gRPC traffic without a prior classification as HTTP2.
+			name: "GRPC without prior HTTP2 classification",
+			context: testContext{
+				serverPort:    http2Port,
+				serverAddress: net.JoinHostPort(serverHost, http2Port),
+				targetAddress: net.JoinHostPort(targetHost, http2Port),
+				extras:        map[string]interface{}{},
+			},
+			preTracerSetup: func(t *testing.T, ctx testContext) {
+				skipIfNotLinux(t, ctx)
+
+				server := NewTCPServerOnAddress(ctx.serverAddress, func(c net.Conn) {})
```

Comment:
```suggestion
				server := NewTCPServerOnAddress(ctx.serverAddress, func(c net.Conn) {c.Close()})
```

Reply:
The issue with this is that, since I'm using `writeInput`, which tries to read a server response, closing the connection in the server handler will make it fail, which makes the test fails.

Reply:
The problem is that you're trying to read but the socket is closed
1. Either don't use `writeInput` as this function expects to read from a socket
2. Or, implement an echo server `io.Copy(c, c); c.Close()`